src/tag/styled-components.js

```diff
@@ -5,10 +5,24 @@ This source code is licensed under the MIT license found in the
 LICENSE file in the root directory of this source tree.
 */
 // @flow
-import {styled, hexToRgb} from '../styles/index';
+import {COLOR_STYLE_KEYS} from './constants';
+import {styled, hexToRgb, type ThemeT} from '../styles/index';
+
+export function getColor(
+  $theme: ThemeT,
+  $color: string,
+  defaultColor: string = '#000',
+) {
+  let themeColor;
+  if ($theme && $theme.colors && $theme.colors[COLOR_STYLE_KEYS.primary]) {
```

Comment:
I think it should be something like 
```
if ($theme && $theme.colors && $theme.colors[$color]) {
  themeColor = $theme.colors[$color];
} 
return $theme.colors[$color] || $color || defaultColor;
```

Reply:
It feels weird to me to overload `color` with named color values from the theme and css color values. Is that a pattern we have anywhere else?

Reply:
We use this pattern in the [icon](https://github.com/uber-web/baseui/blob/master/src/icon/styled-components.js#L13) component.